packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -359,49 +378,127 @@ angular
+                associateIPv6Address: shouldAutoEnableIPv6 || Boolean(ipv6AddressCount),
+                ramdiskId: launchTemplateData.ramdiskId,
+                instanceMonitoring: launchTemplateData.monitoring && launchTemplateData.monitoring.enabled,
+                ebsOptimized: launchTemplateData.ebsOptimized,
+                spotPrice: mip.instancesDistribution.spotMaxPrice || undefined,
+                requireIMDSv2: Boolean(
+                  launchTemplateData.metadataOptions && launchTemplateData.metadataOptions.httpsTokens === 'required',
+                ),
+                unlimitedCpuCredits: launchTemplateData.creditSpecification
+                  ? launchTemplateData.creditSpecification.cpuCredits === 'unlimited'
+                  : undefined,
+                securityGroups: secGroups,
+                onDemandAllocationStrategy: mip.instancesDistribution.onDemandAllocationStrategy,
+                onDemandBaseCapacity: mip.instancesDistribution.onDemandBaseCapacity,
+                onDemandPercentageAboveBaseCapacity: mip.instancesDistribution.onDemandPercentageAboveBaseCapacity,
+                spotAllocationStrategy: mip.instancesDistribution.spotAllocationStrategy,
+                spotInstancePools: mip.instancesDistribution.spotInstancePools,
+              });
+
+              // 'launchTemplateOverridesForInstanceType' is used for multiple instance types case, 'instanceType' is used for all other cases.
+              if (mip.launchTemplateOverridesForInstanceType) {
+                command.launchTemplateOverridesForInstanceType = getInstanceTypesWithPriority(
+                  mip.launchTemplateOverridesForInstanceType,
+                );
+              } else {
+                command.instanceType = launchTemplateData.instanceType;
+              }
 
-            if (serverGroup.launchTemplate && serverGroup.launchTemplate.launchTemplateData.securityGroups.length) {
-              command.securityGroups = serverGroup.launchTemplate.launchTemplateData.securityGroups;
+              command.viewState.imageId = launchTemplateData.imageId;
             }
 
-            if (serverGroup.launchTemplate && serverGroup.launchTemplate.launchTemplateData.networkInterfaces) {
-              const networkInterface =
-                serverGroup.launchTemplate.launchTemplateData.networkInterfaces.find((ni) => ni.deviceIndex === 0) ||
-                {};
-              command.securityGroups = networkInterface.groups;
+            if (mode === 'clone' && serverGroup.image && serverGroup.image.name) {
+              command.amiName = serverGroup.image.name;
             }
 
+            command.viewState.useSimpleInstanceTypeSelector = isSimpleModeEnabled(command);
             return command;
           });
       }
 
+      // Since Deck allows changing priority of instance types via drag handle, fill priority field explicitly if empty
+      function getInstanceTypesWithPriority(instanceTypeOverrides) {
```

Comment:
I have added some tests. Some background and a question for you: 

ASG API doesn't current support assigning explicit priorities to overrides. The priority assigned internally is a contiguous number that relies on the order of overrides in the list. 

In order to support ordering of multiple  instance types / overrides, Deck supports sorting with the drag handle. For this, I have to assign explicit priorities on the first translation of properties into command. 

This function is written to accommodate an explicit  priority, in case the ASG API supports a priority number in the future. I can simplify this function to work for the current case, where a simple increasing priority number is assigned to the items in the list. Let me know what you think.